File: src/main/java/com/jcloisterzone/game/token/FlowersToken.java
package com.jcloisterzone.game.token;

import com.jcloisterzone.game.Token;

public enum FlowersToken implements Token {
    FLOWERS_BLUE("blue"),
    FLOWERS_VIOLET("violet"),
    FLOWERS_WHITE("white"),
    FLOWERS_YELLOW("yellow");

    private final String value;

    FlowersToken(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
    
    public static FlowersToken fromValue(String value) {
        for (FlowersToken token : values()) {
            if (token.value.equals(value)) {
                return token;
            }
        }
        throw new IllegalArgumentException("Unknown flower token value: " + value);
    }
}

File: src/main/ts/com/jcloisterzone/game/token/FlowersToken.ts
import { JavaEnum } from "../../../../lang/JavaEnum.js";
import type { Token } from "../Token.js";

/** Flowers (Village Life): each player is dealt one flower-colour token at game
 *  start; matching flowers on scored features grant +3 each. */
export class FlowersToken extends JavaEnum implements Token {
  static readonly FLOWERS_BLUE = new FlowersToken("FLOWERS_BLUE", 0, "blue");
  static readonly FLOWERS_VIOLET = new FlowersToken("FLOWERS_VIOLET", 1, "violet");
  static readonly FLOWERS_WHITE = new FlowersToken("FLOWERS_WHITE", 2, "white");
  static readonly FLOWERS_YELLOW = new FlowersToken("FLOWERS_YELLOW", 3, "yellow");

  private static readonly VALUES: readonly FlowersToken[] = [
    FlowersToken.FLOWERS_BLUE,
    FlowersToken.FLOWERS_VIOLET,
    FlowersToken.FLOWERS_WHITE,
    FlowersToken.FLOWERS_YELLOW,
  ];

  private constructor(
    name: string,
    ordinal: number,
    private readonly value: string,
  ) {
    super(name, ordinal);
  }

  override toString(): string {
    return this.value;
  }

  static values(): readonly FlowersToken[] {
    return FlowersToken.VALUES;
  }

  static fromValue(value: string): FlowersToken {
    const t = FlowersToken.VALUES.find((v) => v.value === value);
    if (!t) throw new Error("Unknown flowers token " + value);
    return t;
  }
}
